src/interfaces/IERC20.sol

```diff
@@ -175,8 +175,18 @@ interface IERC20Permit {
     function DOMAIN_SEPARATOR() external view returns (bytes32);
 }
 
+struct HookData {
+    uint128 from;
+    uint128 to;
+}
+
 interface IERC20Callback {
-    function onERC20Transfer(address from, address to, uint256 value) external returns (bytes4);
+    function onERC20Transfer(address from, address to, uint256 value, HookData calldata hookdata)
+        external
+        returns (HookData calldata);
+    function onERC20AuthTransfer(address sender, address from, address to, uint256 value, HookData calldata hookdata)
```

Comment:
WHy do we return `HookData` ? Is this something we expect to be changed in any of these callback handlers ?

Reply:
That was the idea, but I realized it does not make sense. Removed!